sdks/go/pkg/beam/core/graph/fn.go

```diff
@@ -304,6 +308,40 @@ func (f *DoFn) PipelineState() []state.PipelineState {
 	return s
 }
 
+type PipelineTimer interface {
+	TimerFamily() string
+	TimerDomain() timers.TimeDomainEnum
+}
```

Comment:
What do you think about switching this to  just a "Timers() map[string]timer.Domain" method, to enable the composites I proposed? https://github.com/apache/beam/issues/25894

We wouldn't need to make the State change in this PR (way way out of scope), but now is ideal for enabling this for timers.